src/goto-instrument/contracts/doc/user/contracts-loops.md

```diff
@@ -168,6 +168,40 @@ void array_wipe(__CPROVER_size_t len, char * array)
 }
 ```
 
+### Caution With Nested Loop
+Due to the nature of @ref contracts-assigns, we need to be aware of the non-deterministic value of the assigned target.
+In the example below, 
+```c
+const unsigned table[256] = {1, 2, 3, ..., 256};
+
+void nested_loop_example() {
+        unsigned t = 0;
+        for( j=0; j<18; j++ )
+        __CPROVER_assigns(t, j, k) // t and k need to be included in the outer loop
+        __CPROVER_loop_invariant(0 <= j && j <= 18 && t < 256) // t < 256 must be included in the outer loop invariant
+        __CPROVER_decreases(18 - j)
+        {
+            /*
+              during instrumentation, the assigns clause is equivalent to:
+              __CPROVER_havoc(t);
+              __CPROVER_havoc(j);
+            */
```

Comment:
```suggestion
```
I don't think users really need to understand the idea of havocing. Instead, we should say that the inductive step is based on an arbitrary state of the assigns targets.